### PR TITLE
vim-patch:8.2.0212: missing search/substitute pattern hardly tested

### PR DESCRIPTION
**Pull request by VVKot**
Problem:    Missing search/substitute pattern hardly tested.
Solution:   Add test_clear_search_pat() and tests. (Yegappan Lakshmanan,
            closes vim/vim#5579)
https://github.com/vim/vim/commit/07ada5ff2fd8f22ed3233ae5c4ddf87c7b3f56fe

### CODE REVIEW

**Comment by VVKot** [comment]
Pending response in #15187

File: src/nvim/testdir/test_substitute.vim [hunk 1]
```diff
@@ -765,4 +765,20 @@ func Test_substitute_skipped_range()
   bwipe!
 endfunc
 
+" Test for command failures when the last substitute pattern is not set.
+func Test_sub_with_no_last_pat()
+  " TODO how to override last substitution pattern?
```
Comment (VVKot) [inline src/nvim/testdir/test_substitute.vim:770]: Thoughts here?

Comment (VVKot) [reply src/nvim/testdir/test_substitute.vim:770]: Ah anyhow, they later remove it https://github.com/vim/vim/commit/4f5776c17cd86f904a7e2f92db297c73e28939b7 - so I'll port up related patches up to that one.

File: src/nvim/testdir/test_vimscript.vim [hunk 2]
```diff
@@ -1684,6 +1684,92 @@ func Test_function_defined_line()
...
+  call writefile(['while v:true', 'echo "."'], 'Xscript')
+  call assert_fails('source Xscript', 'E170:')
+  call writefile(['try', 'echo "."'], 'Xscript')
+  call assert_fails('source Xscript', 'E600:')
+  call delete('Xscript')
+endfunc
+
+" Test for deep nesting of if/for/while/try statements              {{{1
+func Test_deep_nest()
+  if !CanRunVimInTerminal()
```
Comment (VVKot) [inline src/nvim/testdir/test_vimscript.vim:1702]: Seems this needs `source term_util.vim`.

File: src/nvim/testdir/test_ex_mode.vim [hunk 3]
```diff
@@ -98,4 +98,26 @@ func Test_ex_mode_count_overflow()
...
+func Test_open_command()
+  new
+  call setline(1, ['foo foo', 'foo bar', 'foo baz'])
+  call feedkeys("Qopen\<CR>j", 'xt')
+  call assert_equal('foo bar', getline('.'))
+  call feedkeys("Qopen /bar/\<CR>", 'xt')
+  call assert_equal(5, col('.'))
+  call assert_fails('call feedkeys("Qopen /baz/\<CR>", "xt")', 'E479:')
+  close!
+endfunc
```
Comment (VVKot) [inline src/nvim/testdir/test_ex_mode.vim:121]: No idea what's wrong with this test.

File: src/nvim/testdir/test_window_cmd.vim [hunk 4]
```diff
@@ -924,4 +924,18 @@ func Test_window_resize()
...
+  call assert_fails('wincmd ab', 'E474:')
+endfunc
+
+" Test for errors with :winpos
+func Test_winpos_errors()
+  if !has("gui_running") && !has('win32')
+    call assert_fails('winpos', 'E188:')
+  endif
+  call assert_fails('winpos 10', 'E466:')
+endfunc
```
Comment (VVKot) [inline src/nvim/testdir/test_window_cmd.vim:939]: Is `winpos` purposefully not included in nvim or it just wasn't ported?

**Comment by VVKot** [comment]
@janlazo wonder if you can take a look?

File: src/nvim/testdir/test_window_cmd.vim (hunk 4 above)
Comment (janlazo) [reply src/nvim/testdir/test_window_cmd.vim:939]: https://github.com/neovim/neovim/commit/e0e41b30#diff-b22503b06d373fd31e6496ab2791935466d123f9b69fa2c62a89d9cf510c0d97R2486

File: src/nvim/testdir/test_cd.vim [hunk 5]
```diff
@@ -46,7 +46,6 @@ func Test_cd_minus()
   " Test for :cd - without a previous directory
   let lines =<< trim [SCRIPT]
     call assert_fails('cd -', 'E186:')
-    call assert_fails('call chdir("-")', 'E186:')
```
Comment (janlazo) [inline src/nvim/testdir/test_cd.vim]: Comment. Not delete. If the test fails, then port chdir.

**Pull request closed without merging**